partner_credit_limit/model/sale.py

```diff
@@ -8,10 +8,16 @@
 class SaleOrder(models.Model):
     _inherit = 'sale.order'
 
+    @api.multi
+    def check_payment_type(self):
+        if self.filtered(lambda so:
+                         so.payment_term_id.payment_type != 'credit'):
+            return True
+
     @api.multi
     def check_limit(self):
         for so in self:
-            if so.payment_term_id.payment_type != 'credit':
+            if so.check_payment_type():
```

Comment:
could you use filtered into for?

Reply:
I don't understand how can I use filtered here.

Reply:
`self.filtered(lambda inv: inv.check_payment_type()):`